agents/gemini_data_analytics/intro_gemini_data_analytics_sdk.ipynb

```
@@ -0,0 +1,1574 @@
+        "\n",
+        "def display_datasource(datasource):\n",
+        "    source_name = \"\"\n",
+        "    if \"studio_datasource_id\" in datasource:\n",
+        "        source_name = datasource.studio_datasource_id\n",
+        "    elif \"looker_explore_reference\" in datasource:\n",
+        "        source_name = format_looker_table_ref(datasource.looker_explore_reference)\n",
+        "    else:\n",
+        "        source_name = format_bq_table_ref(datasource.bigquery_table_reference)\n",
+        "\n",
+        "    print(source_name)\n",
+        "    display_schema(datasource.schema)\n",
```

Ruff (0.12.2)

Possible hardcoded password assigned to: "looker_client_secret"

(S105)